### ISSUE TITLE
[BUG] Updating a repository already in use throws 500

### ISSUE THREAD

**Issue by imRishN** [opensearch-project/OpenSearch#7336]
**Describe the bug**
Currently, when we try to update or unregister a repository which is already in use by one of the snapshots throws 500 because the underlying exception is `IllegalStateException`.

```
private static void ensureRepositoryNotInUse(ClusterState clusterState, String repository) {
        if (isRepositoryInUse(clusterState, repository)) {
            throw new IllegalStateException("trying to modify or unregister repository that is currently used");
        }
    }
```

Should we modify the exception here and instead of 500 should we throw 403 status code to the user?

Stack trace - 

```
[2023-04-29T13:12:28,402][WARN ][r.suppressed             ] [<id>] path: /_snapshot/<repo name>, params: {repository=<repo name>}
RemoteTransportException[[<id>][<ip>][cluster:admin/repository/put]]; nested: IllegalStateException[trying to modify or unregister repository that is currently used];
Caused by: java.lang.IllegalStateException: trying to modify or unregister repository that is currently used
        at org.opensearch.repositories.RepositoriesService.ensureRepositoryNotInUse(RepositoriesService.java:633)
        at org.opensearch.repositories.RepositoriesService$1.execute(RepositoriesService.java:202)
        at org.opensearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:65)
        at org.opensearch.cluster.service.MasterService.executeTasks(MasterService.java:874)
        at org.opensearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:424)
        at org.opensearch.cluster.service.MasterService.runTasks(MasterService.java:295)
        at org.opensearch.cluster.service.MasterService$Batcher.run(MasterService.java:206)
        at org.opensearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:190)
        at org.opensearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:228)
        at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:756)
        at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:282)
        at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:245)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(Thread.java:829)
[2023-04-29T13:12:28,402][INFO ][c.a.c.e.logger           ] [<id>] PUT /_snapshot/<repo name> - 500 INTERNAL_SERVER_ERROR 259 3
```

**To Reproduce**

**Expected behavior**

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

**Comment by dbwiddis** [comment]
@imRishN will you be able to contribute a fix?  I agree with @dblock that returning a 409 with a more useful message is the right approach.

**Comment by Ayan07** [comment]
@dbwiddis / @dblock I would like to contribute to this project.
Can you assign this issue to me if this is something that can still be worked upon?
Thanks!

**Comment by aibarelemesov** [comment]
@dbwiddis / @dblock Id like to work on this issue.
did some research and if I'm not mistaken the status code for ensureRepositoryNotInUse comes from [here](https://github.com/opensearch-project/OpenSearch/blob/5fdebeef571c5fa8b2daf641ccffa895d3b2446d/libs/core/src/main/java/org/opensearch/ExceptionsHelper.java#L94) . Should I bind IllegalStateException to RestStatus.CONFLICT in this method or create a new exception for RestStatus.CONFLICT to avoid changing the status code for IllegalStateException globally?

**Comment by dblock** [comment]
Is this the same as https://github.com/opensearch-project/OpenSearch/pull/8986? @baba-devv did your PR resolve this exact issue?

**Comment by baba-devv** [comment]
@dblock the bug that my PR resolved stated "amazon_s3_exception: The specified bucket does not exist", so it was for when the queried bucket didn't exist, but I reckon that change will resolve this issue as well since that was for "Concurrent Snapshot Execution".

**Comment by dblock** [comment]
Thanks @baba-devv. I'm closing this as fixed by #8986.
